### PR TITLE
Secure application

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket 107020530](https://www.pivotaltracker.com/story/show/107020530)

### CODE REVIEW

**Comment by arothuis** [comment]
@rjkip, @DRvanR

File: app/config/config.yml [hunk 1]
```diff
@@ -26,6 +26,44 @@ framework:
...
+            '^/.*': DENY
+    external_redirects:
+        abort: true
+        log: true
+        whitelist:
+            - %saml_remote_idp_host%
+
+    csp:
+        report:
+            report-uri: [/nelmio/csp/report]
```
Comment (rjkip) [inline app/config/config.yml]: Do you have a controller set up for these CSP reports (see links below)? Can there be multiple report URLs? Also, do we only enforce `script-src: [self]` (JavaScript)? What about `default-src: [self]` (everything)?

https://github.com/SURFnet/Stepup-SelfService/blob/d415fc9e81d1b0663a85f867b10ce8461345aee3/app/config/config.yml#L71
https://github.com/SURFnet/Stepup-SelfService/blob/1ba8b776aff73aaa490058420de06d5e105c0206/app/config/routing.yml#L6

File: src/OpenConext/ProfileBundle/Security/Guard.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
+    public function userIsLoggedIn()
+    {
+        if (!$this->authorizationChecker->isGranted('ROLE_USER')) {
+            throw new AccessDeniedHttpException('User should be logged in to use this service');
```
Comment (rjkip) [inline src/OpenConext/ProfileBundle/Security/Guard.php]: I think we should either throw a [AccessDeniedException](https://github.com/symfony/symfony/blob/0055887cbdf89d9b3b0235f2089a2751bcb52d17/src/Symfony/Component/Security/Core/Exception/AccessDeniedException.php), which Symfony [converts](https://github.com/symfony/symfony/blob/0055887cbdf89d9b3b0235f2089a2751bcb52d17/src/Symfony/Component/Security/Http/Firewall/ExceptionListener.php#L119) to an AccessDeniedHttpException, or move the Guard to the namespace `...\Security\Http`.

File: app/config/config.yml (hunk 1 above)
Comment (DRvanR) [inline app/config/config.yml]: `none` for object and `[self]` for the rest would be better indeed :stuck_out_tongue_winking_eye: If we need something different in dev, we can configure that there.

And good point about the CSP enpoint, the report uri currently used is the default nelmio provided one. Would like to have the linked logging version though.

File: src/OpenConext/ProfileBundle/Security/Guard.php (hunk 2 above)
Comment (DRvanR) [inline src/OpenConext/ProfileBundle/Security/Guard.php]: The former indeed, `AccessDeniedException`.

**Comment by arothuis** [comment]
@rjkip, @DRvanR

**Comment by rjkip** [comment]
:+1:

**Comment by DRvanR** [comment]
nice nice :+1:

**Pull request merged**